packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,129 @@
+/*
+- This plugin will create a master svg with two child svg's.
+- One dark and one light.
+- The children svgs will have a `display` attribute that can be used to set
+which of the two svg's will be visible. */
+
+import {
+  Flavor,
+  GeneratePlugin,
+  Icon
+} from '@icon-magic/icon-models';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import Svgo from 'svgo';
+import { DOMImplementation, DOMParser } from 'xmldom';
+
+
+export interface SvgLightDarkOptions {
+  lightToken?: string;
+  darkToken?: string;
+}
+
+export const svgLightDark: GeneratePlugin = {
+  name: 'svg-light-dark',
+  fn: async (
+    flavor: Flavor,
+    icon: Icon,
+    params: SvgLightDarkOptions = {},
+  ): Promise<Flavor> => {
+    // Create the output directory
+    const outputPath = icon.getIconOutputPath();
+    const imageSet = flavor.imageset;
+
+
+    // Only dark flavors have an imageSet Attribute
+    // If imageSet exists, plugin will run (on the "dark" flavor only)
+    // plugin will skip the "light" flavors because their data can be gotten
+    // with the dark flavor
+    if (imageSet) {
+      const darkFlavorContents = (await flavor.getContents()) as string; // .svg asset's getContents() returns a string (the svg as a string)
+      const lightFlavor = icon.flavors.get(imageSet);
+      let lightFlavorContents;
+      if (lightFlavor) {
```

Comment:
if the light flavor doesn't exist we shouldn't move forward with the process at all? I think you should add a return statement if the light flavor doesn't exist.

Reply:
What do you think about this:
```
      // if light flavor does not exist return, else proceed with plugin
      if (lightFlavor!) {
        return flavor
      } else {

      }
   ```
   
   having the plugin run in the else statement?